pkg/scraper/client/resource/decode_test.go

```diff
@@ -241,3 +242,34 @@ container_start_time_seconds{container="metrics-server",namespace="kubernetes-da
 		})
 	}
 }
+
+func Fuzz_decodeBatch(f *testing.F) {
+	emptyMetrics := &storage.MetricsBatch{
+		Nodes: map[string]storage.MetricsPoint{},
+		Pods:  map[apitypes.NamespacedName]storage.PodMetricsPoint{},
+	}
+	f.Fuzz(func(t *testing.T, cpuValue float64, memValue float64, startTimeValue float64, timeStamp int64, defaultTimeValue int64) {
+		defaultTime := time.Unix(0, defaultTimeValue)
+		input := fmt.Sprintf(
+			`# HELP container_cpu_usage_seconds_total [ALPHA] Cumulative cpu time consumed by the container in core-seconds
```

Comment:
Testing something that resembles prometheus input is a good idea, however we should also check how if MS panics on invalid data.

Reply:
ok let me add more scenes